### PR TITLE
pyflyte run can now execute a task either locally or remote

### PR DESCRIPTION
**Pull request by kumare3**
Signed-off-by: Ketan Umare <ketan.umare@gmail.com>

# TL;DR
It is now possible to execute a single task from the command line

Example
```python
from flytekit import workflow, task
from typing import List, Dict

@task
def t1(l: str) -> List[str]:
    return [l]


@workflow
def wf(l: str = "") -> List[str]:
    return t1(l=l)


@task
def t2(i: Dict[str, int]) -> Dict[str, int]:
    return i

@workflow
def wf2m(i: Dict[str, int]) -> Dict[str, int]:
    return t2(i=i)
```
*To Execute*
```
pyflyte run --remote test2.py t1 --l hello
```

and help should show
```bash
➜ pyflyte run test2.py --help

Usage: pyflyte run test2.py [OPTIONS] COMMAND [ARGS]...

  Run a [workflow|task] in a file using script mode

Options:
  --help  Show this message and exit.

Commands:
  wf    Run test2.wf in script mode
  wf2m  Run test2.wf2m in script mode
  t1    Run test2.t1 in script mode
  t2    Run test2.t2 in script mode
```

## Type
 - [ ] Bug Fix
 - [x] Feature
 - [ ] Plugin

## Are all requirements met?

 - [x] Code completed
 - [x] Smoke tested
 - [x] Unit tests added
 - [x] Code documentation added
 - [x] Any pending items have an associated Issue

## Tracking Issue
https://github.com/flyteorg/flyte/issues/2471

## Follow-up issue
_NA_

### CODE REVIEW

File: flytekit/clis/sdk_in_container/run.py [hunk 1]
```diff
@@ -422,9 +423,9 @@ def dump_flyte_remote_snippet(execution: FlyteWorkflowExecution, project: str, d
     )
 
 
-def get_workflows_in_file(filename: str) -> typing.List[str]:
+def get_entities_in_file(filename: str) -> typing.Tuple[typing.List[str], typing.List[str]]:
```
Comment (samhita-alla) [inline flytekit/clis/sdk_in_container/run.py:426]: Can we not return a single list of workflows and tasks cause we're anyways combining them in `list_commands` as follows:

```python
workflows, tasks = get_entities_in_file(self._filename)
workflows.extend(tasks)
```

Comment (samhita-alla) [reply flytekit/clis/sdk_in_container/run.py:426]: Or better, return a dictionary of lists to keep it clean, say: `{"workflows": [...],  "tasks": [...]}`

**Review by samhita-alla — APPROVED** [review]

File: flytekit/clis/sdk_in_container/run.py [hunk 2]
```diff
@@ -513,18 +533,18 @@ def __init__(self, filename: str, *args, **kwargs):
         self._filename = filename
 
     def list_commands(self, ctx):
-        workflows = get_workflows_in_file(self._filename)
-        return workflows
+        entities = get_entities_in_file(self._filename)
+        return entities.all()
 
-    def get_command(self, ctx, workflow):
+    def get_command(self, ctx, exe_entity):
```
Comment (katrogan) [inline flytekit/clis/sdk_in_container/run.py:539]: what's exe_entity?

**Review by katrogan — APPROVED** [review]

File: flytekit/clis/sdk_in_container/run.py (hunk 2 above)
Comment (kumare3) [reply flytekit/clis/sdk_in_container/run.py:539]: executable

**Review by wild-endeavor — APPROVED** [review]

**Pull request merged**